### PR TITLE
[CPU] Support group beam search

### PR DESCRIPTION
**Pull request by luo-cheng2021**
### Details:
 - *Support group beam search*
 - *...*

### Tickets:
 - *128111*
 - *128815*

### CODE REVIEW

**Review by usstq — APPROVED** [review]
LGTM, thanks!

File: src/plugins/intel_cpu/src/nodes/scaled_attn.cpp [hunk 1]
```diff
@@ -822,13 +822,158 @@ void ScaledDotProductAttention::assignState(const std::shared_ptr<VariableStateK
...
+        new_pastv = new_pastv.permute(order);
+        if (L0 > 0) {
+            auto old_internal_mem_k = m_k_state->internal_state_mem();
+            auto old_internal_mem_v = m_v_state->internal_state_mem();
+            old_past_k.reset(old_internal_mem_k);
+            old_past_v.reset(old_internal_mem_v);
+            old_past_k = old_past_k.permute(order);
+            old_past_v = old_past_v.permute(order);
+            parallel_for3d(B, H, L0, [&](size_t b, size_t h, size_t m) {
+                auto b_kv = static_cast<size_t>(cur_beam_table.at<int32_t>({b, m}));
```
[the remark this answers is not in the document]
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/nodes/scaled_attn.cpp]: Done.

**Review by dmitry-gorokhov — APPROVED** [review]
In general LGTM. However we need to consider some refactoring as follow-up activity: `resetBeamTablePastkv`, `updatePastkv` and `updateBeamTable` now contains quite a lot of duplicated logic, so that part need some improvements.
@luo-cheng2021  @usstq

**Comment by yuxu42** [comment]
@dmitry-gorokhov can we merge the PR? Thanks!

**Pull request merged**